third-party/gasnet/gasnet-src/gasnet_internal.c

```diff
@@ -2129,7 +2129,7 @@ extern void gasneti_nodemapParse(void) {
     }
   #endif
   
-#if GASNET_NDEBUG && !GASNET_PSHM && !GASNET_SEGMENT_EVERYTHING
+#if GASNET_NDEBUG && !GASNET_PSHM && !GASNET_SEGMENT_EVERYTHING && !GASNET_NO_PSHM_WARNING
```

Comment:
Typically we don't like changing third-party code without a plan to upstream it. In this case I'd pull this out and open up a bugzilla entry with the gasnet team to request a way to quiet this warning.

Reply:
If we don't quiet the warning it isn't feasible to run more than one locale on a node as you always get the warning in the output. My plan is to ask `gasnet` to add a way to quiet it, but do this in the meantime.

Reply:
Not feasible how? Like testing will fail? If it's just about testing, we can use a system prediff

Reply:
I should have said "practical" rather than "feasible". With the warning enabled we'd have to do a system-wide prediff to remove it from test outputs. I'm guessing that isn't too difficult, although I have to look into how that is done. The bigger problem to  me is that the users get the following warning every time they run with co-locales:
```
> ./hello -nl 2                                                                                                                                         gas-lps
*** WARNING (proc 0): Running with multiple processes per host without shared-memory communication support (PSHM).  This can significantly reduce performance.  Please re-configure GASNet using `--enable-pshm` to enable fast intra-host comms.
Hello, world!
```
They are immediately going to complain to us and there isn't anything they can do to turn it off.

Reply:
👍

Reply:
I reverted this change.